### ISSUE TITLE
Issue with UI clipping on scale factor greater than 1

### ISSUE THREAD

**Issue by mockersf** [bevyengine/bevy#3493]
## Bevy version

current main d8d3fdf2cfa53c511e8e8156fe1c8a749a20f159

## Operating system & version

macOS & webgl2

## What you did

Run `ui` example at different scales

## What you expected to happen

Clipping should work the same

## What actually happened

Clipping has artefacts at scale greater than 1. They are more visible at the bottom of the list

scale 1:
<img width="752" alt="Screenshot 2021-12-30 at 04 09 41" src="https://user-images.githubusercontent.com/8672791/147718773-d771606d-8193-412a-a9a9-eb4a10392549.png">

scale 2:
<img width="1392" alt="Screenshot 2021-12-30 at 04 07 09" src="https://user-images.githubusercontent.com/8672791/147718777-8311a557-231a-4278-94d9-d7de73d97d18.png">

scale 3:
<img width="1912" alt="Screenshot 2021-12-30 at 04 10 51" src="https://user-images.githubusercontent.com/8672791/147718779-f4b31ee1-6101-4fc6-bbb8-8305b2a93e80.png">

## Additional information

You can add
```rust
        .insert_resource(WindowDescriptor {
            scale_factor_override: Some(3.0),
            ..Default::default()
        })
```
to the ui example to make it visible

**Comment by Davier** [comment]
Found the issue, the node size and atlas size need to be scaled in some equations. I'm working on a fix.

**Comment by Weibye** [comment]
Was in the process of adopting #3533 but realized the problem had already been fixed by #3854
